pkg/trace/api/api.go

```diff
@@ -141,7 +145,7 @@ func (r *HTTPReceiver) Listen(addr, logExtra string) error {
 
 	go func() {
 		defer watchdog.LogOnPanic()
-		ln.Refresh(r.conf.ConnectionLimit)
+		ln.Refresh(r.conf.ConnectionLimit, r.exit)
```

Comment:
Even though this is a nice shortcut, it would make more sense for the `rateLimitedListener` to manage its own routines. It could then be signaled inside a new `Close` method which would override the [wrapped `*net.TCPListener`'s](https://github.com/DataDog/datadog-agent/blob/41c60b3193f2e9bc829188158d21cfd71c268392/pkg/trace/api/listener.go#L15) method.

When the server shuts down, that `Close` method will be [called automatically](https://github.com/golang/go/blob/master/src/net/http/server.go#L2844). I'd imagine it would look something like this:

```go
func (rln *rateLimitedListener) Close() error {
    rln.exit <- struct{}{}
    <-rln.exit
    return rln.TCPListener.Close()
}
```
What do you think?